Src/ILGPU/Backends/EntryPoints/EntryPointDescription.cs

```diff
@@ -56,7 +56,7 @@ namespace ILGPU.Backends.EntryPoints
             // Try to get index type from first parameter
             var firstParamType = parameters[0].ParameterType;
             var indexType = firstParamType.GetIndexType();
-            if (indexType == IndexType.None)
+            if (indexType == IndexType.None || indexType > IndexType.Index3D)
```

Comment:
QUESTION: This excludes `LongIndexX` from being used as the first parameter of a kernel. Is this the intended behaviour?

```suggestion
            if (indexType < IndexType.Index1D || indexType > IndexType.LongIndex3D)
```

Reply:
Currently, this patch does not include support for implicitly grouped kernels that use the `LongIndexX` types. If we add this convenience feature, we can remove these checks and accept all compatible intrinsic index types.

Reply:
Not critical 👍